tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/ModifyRepositoryMetadataMojo.java

```diff
@@ -0,0 +1,151 @@
+/*******************************************************************************
+ * Copyright (c) 2025, 2025 Hannes Wellmann and others.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Hannes Wellmann - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.tycho.plugins.p2.repository;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.maven.model.Repository;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.plexus.logging.Logger;
+import org.eclipse.equinox.p2.core.ProvisionException;
+import org.eclipse.tycho.p2maven.repository.P2RepositoryKind;
+import org.eclipse.tycho.p2maven.tools.P2RepositoryDataManipulator;
+import org.eclipse.tycho.p2maven.tools.P2RepositoryDataManipulator.ModifiedRepositoryDescriptor;
+
+/**
+ * Modifies metadata of a P2-repository like its name or properties.
+ * <p>
+ * This can be used to customize metadata of a just assembled repository or to modify an already
+ * existing (remote) repository. The modified repository metadata files
+ * ({@code artifact.xml/jar/xml.xz} or {@code content.xml/jar/xml.xz}) are always written to a local
+ * file. For a remote repository a suitable other tool (like SSH) must be used to copy these files
+ * subsequently back to the remote server.
+ * </p>
+ */
+@Mojo(name = "modify-metadata", threadSafe = true)
```

Comment:
Should we better name it


```suggestion
@Mojo(name = "modify-p2-metadata", threadSafe = true)
```

?

Reply:
The mojo is in `tycho-p2-repository-plugin` and other mojos operating on p2-repos don't mention p2 explicitly.
On the other hand, this mojo is about modifying the metadata of the repository itself (and the repository can be a `metadata` repository). So mentioning `p2-metadata` could help to avoid confusion with a mojo that would modify just a metadata repository (which this doesn't).
So I renamed it as you suggested.

Reply:
On the other hand, with that in mind, it might be even better to name it `modify-repository-metadata`?

Reply:
> On the other hand, with that in mind, it might be even better to name it `modify-repository-metadata`?

Why not simply `modify-repository` ?

Reply:
This sounds very general. And while we can for sure extend the possible modifications in the future, I'm not sure if it will and should evolve into a mojo that's so general that such general name is really suitable.

Reply:
As you mentioned there is artifact and metadata repository. and as both are metadata each moidification of a repo include the metadata itself.

So I think it is okay here if we really want it to modify both types, otherwise an abstract base mojo and two specializations as

- `modify-metadata-repository`
- `modify-artifact-repository`

would be more suitable.

Reply:
As discussed in a private chat, I changed the name now to `modify-repository-properties` as that's exactly what the mojo is doing (and the name is also a property).